src/components/SignupPage/SignUp.js

```diff
@@ -21,6 +49,26 @@ export function SignUp({ setSessionObject } ) {
         data.familyName = formProps.signUpLastName;
         data.address = formProps.address;
         data.role = 'user';
+        data.confirmPass = formProps.signUpConfirmPassword;
+
+        if (data.password.length < minNumberofChars || data.password.length > maxNumberofChars) {
+            setErrorMessage('The length of the Password should be between 8 and 16 characters');
+            setState(false);
+        } if (data.dateOfBirth > today) {
+            setErrorMessage('Birthday should be in the past')
+            console.log(data.dateOfBirth, today)
+            setState(false)
+        } else if (data.password !== data.confirmPass) {
+            setErrorMessage('!!! Passwords do not match');
+            setState(false);
+        } else if (!regularExpression.test(data.password)) {
+            setErrorMessage('Password should contain at least one uppercase letter, one lowercase letter, and one number');
+            setState(false);
+        } else {
+            setErrorMessage('You signed up!');
+            setState(true);
+        }
```

Comment:
This should be moved to happen after singUpAdapter is called (inside of then)

Reply:
moved inside .then as suggested